src/runtime/flags.ts

```diff
@@ -12,6 +12,7 @@
 
 class FlagDefaults {
   static useNewStorageStack = false;
+  static enforceRefinements = true;
```

Comment:
We may have to turn this off by default if we find issues with the implementation. Probably safer to start with it off (so that our tests for its behavior have to turn if on explicitly) and then feel free to turn it on or off at will later.